reactive/docker.py

```diff
@@ -47,56 +47,79 @@
 # Be sure you bind to it appropriately in your workload layer and
 # react to the proper event.
 
-dockerpackages = {'apt': ['docker.io'],
-                  'upstream': ['docker-engine'],
-                  'nvidia': ['docker-ce',
-                             'nvidia-docker2',
-                             'nvidia-container-runtime',
-                             'nvidia-container-runtime-hook']}
+docker_packages: dict = {
+    'apt': ['docker.io'],
+    'upstream': ['docker-engine'],
+    'nvidia': [
+        'docker-ce',
+        'nvidia-docker2',
+        'nvidia-container-runtime',
+        'nvidia-container-runtime-hook'
+    ]
+}
 
 
-def holdall():
-    for k in dockerpackages.keys():
-        apt_hold(dockerpackages[k])
+def hold_all() -> None:
+    """
+    Hold packages.
 
+    :return: None
+    """
+    for k in docker_packages.keys():
+        apt_hold(docker_packages[k])
 
-def unholdall():
-    for k in dockerpackages.keys():
-        apt_unhold(dockerpackages[k])
 
+def unhold_all() -> None:
+    """
+    Unhold Packages.
 
-@hook('upgrade-charm')
-def upgrade():
-    holdall()
-    hookenv.log('Holding docker packages' +
-                ' at current revision.')
+    :return: None
+    """
+    for k in docker_packages.keys():
+        apt_unhold(docker_packages[k])
 
 
-def determineAptSource():
+@hook('upgrade-charm')
+def upgrade() -> None:
+    """
+    :return: None
+    """
+    hold_all()
+    hookenv.log(
+        'Holding docker packages at current revision.')
+
+
+def determine_apt_source() -> str:
+    """
+    :return: String docker runtime
+    """
     docker_runtime = config('docker_runtime')
 
     if config('install_from_upstream'):
         docker_runtime = 'upstream'
 
-    if docker_runtime == "auto":
+    if docker_runtime == 'auto':
         out = check_output(['lspci', '-nnk']).rstrip()
         if arch() == 'amd64' \
-                and out.decode('utf-8').lower().count("nvidia") > 0:
-            docker_runtime = "nvidia"
+                and out.decode('utf-8').lower().count('nvidia') > 0:
+            docker_runtime = 'nvidia'
         else:
-            docker_runtime = "apt"
+            docker_runtime = 'apt'
 
-    hookenv.log('Setting runtime to {0}'.format(docker_runtime))
+    hookenv.log(f'Setting runtime to {docker_packages}')
```

Comment:
s/docker_packages/docker_runtime/

Also don't f-strings require python 3.6, which would not be the default on xenial...